raven/contrib/tornado/__init__.py

```diff
@@ -8,11 +8,27 @@
 from __future__ import absolute_import
 
 import time
+import uuid
+import zlib
+import base64
+import datetime
 
 import raven
 from raven.base import Client
-from raven.utils import get_auth_header
+from raven.utils import json, get_auth_header
 from tornado.httpclient import AsyncHTTPClient, HTTPError
+from tornado.concurrent import Future
+
+
+class TornadoJSONEncoder(json.BetterJSONEncoder):
+    ENCODER_BY_TYPE = {
+        uuid.UUID: lambda o: o.hex,
+        datetime.datetime: lambda o: o.strftime('%Y-%m-%dT%H:%M:%SZ'),
+        set: list,
+        frozenset: list,
+        bytes: lambda o: o.decode('utf-8', errors='replace'),
+        Future: lambda o: None,
```

Comment:
Perhaps you could do something slightly more explicit about Futures rather than just None.
python```
if future.resolved():
    return yield from future
else:
    return future.__repr__()
```

Other than that :shipit: